### ISSUE TITLE
Reorganise project branches and workflows

### ISSUE THREAD

**Issue by vaadin-miki** [vaadin-miki/super-fields#100]
The following structure should work:
* `master` should only be used for released versions, it is expected to be stable and safe to use
* `development` should be main branch into which code goes, all tests pass, but the behaviour may be quirky; it has the next version + snapshot modifier in it
* `release-X.Y.Z` should branch off of `development`, targetting `master`, when the release is ready; upon creating that branch all tests must be done, but before that happens a workflow should set versions of `superfields` and `superfields-parent` to whatever `X.Y.Z` is; once a pr is merged into master, if possible, a tag should be made automatically to it, so that a new release is done

Ticket branches should still start with a ticket number, followed by `-scope` of work. More than one ticket can be worked on at the same time in such branch. If it is known at the time of starting work, branch name should be `A-B-scope` (where A, B, are ticket numbers).

If automation is not possible to the degree described above, keep the current model.

Also, create a wiki page explaining the branches and how to contribute to this project.
